File: data-pipeline-flink/dp-telemetry-reader/src/main/java/org/sunbird/dp/reader/ParentMap.java
package org.sunbird.dp.reader;

import org.sunbird.dp.util.Logger;

import java.util.Map;

// import org.ekstep.ep.samza.core.Logger;

class ParentMap implements ParentType {
    static Logger LOGGER = new Logger(ParentMap.class);
    Map<String, Object> map;
    String childKey;

    ParentMap(Map<String, Object> map, String childKey) {
        this.map = map;
        this.childKey = childKey;
    }

    @Override
    public <T> T readChild() {
        if (map != null && map.containsKey(childKey) && map.get(childKey) != null) {
            Object child = map.get(childKey);
            return (T) child;
        }
        return null;
    }

    @Override
    public void addChild(Object value) {
        if (map != null)
            map.put(childKey, value);
    }
}

File: data-pipeline-flink/dp-core/src/main/scala/org/sunbird/dp/core/reader/ParentMap.scala
package org.sunbird.dp.core.reader


import java.util

class ParentMap private[reader](var map: util.Map[String, Any], var childKey: String) extends ParentType {
  override def readChild[T]: Option[T] = {
    if (map != null && map.containsKey(childKey) && map.get(childKey) != null) {
      val child = map.get(childKey)
      return Some(child.asInstanceOf[T])
    }
    None
  }

  override def addChild(value: Any): Unit = {
    if (map != null) map.put(childKey, value)
  }
}
